rdtools/normalization.py

```diff
@@ -1,63 +1,125 @@
-''' Energy Normalization Module
-
-This module contains functions to help normalize AC energy output with measured
-poa_global in preparation for calculating PV system degradation.
-'''
+'''Functions for normalizing, rescaling, and regularizing PV system data.'''
 
 import pandas as pd
 import pvlib
 import numpy as np
 from scipy.optimize import minimize
 import warnings
-
+from rdtools._deprecation import deprecated
 
 class ConvergenceError(Exception):
+    '''Rescale optimization did not converge'''
     pass
 
+def normalize_with_expected_power(pv, power_expected, poa_global,
+                                  pv_input='power'):
+    '''
+    Normalize pv output based on expected PV power.
+
+    Parameters
+    ----------
+    pv : pd.Series
+        Right-labeled time series PV energy or power. If energy, should *not*
+        be cumulative, but only for preceding time step.
+    power_expected : pd.Series
+        Right-labeled time series of expected PV power.
+    poa_global : pd.Series
+        Right-labeled time series of plane-of-array irradiance associated with
+        `expected_power`
+    pv_input : str
+        'power' or 'energy' to specify type of input used for pv parameter
+
+    Returns
+    -------
+    energy_normalized : pd.Series
+        Energy normalized based on `expected_power`
+    insolation : pd.Series
+        Insolation associated with each normalized point
 
-def pvwatts_dc_power(poa_global, P_ref, T_cell=None, G_ref=1000, T_ref=25, gamma_pdc=None):
+    '''
+
+    freq = check_series_frequency(pv, 'pv')
+
+    if pv_input == 'power':
+        energy = energy_from_power(pv, freq)
+    elif pv_input == 'energy':
+        energy = pv.copy()
+        energy.name = 'energy_Wh'
+    else:
+        raise ValueError("Unexpected value for pv_input. pv_input should be 'power' or 'energy'.")
+
+    model_tds, mean_model_td = delta_index(power_expected)
+    measure_tds, mean_measure_td = delta_index(energy)
+
+    # Case in which the model less frequent than the measurements
+    if mean_model_td > mean_measure_td:
+        power_expected = interpolate(power_expected, pv.index)
+        poa_global = interpolate(poa_global, pv.index)
+
+    energy_expected = energy_from_power(power_expected, freq)
+    insolation = energy_from_power(poa_global, freq)
+
+    energy_normalized = energy / energy_expected
+
+    index_union = energy_normalized.index.union(insolation.index)
+    energy_normalized = energy_normalized.reindex(index_union)
+    insolation = insolation.reindex(index_union)
+
+    return energy_normalized, insolation
+
+
+@deprecated(since='2.0.0', removal='3.0.0',
+            alternative='normalize_with_expected_power')
```

Comment:
Do we have a function to generate `expected_power` when this function is removed? I would like to encourage the team to include the same range of expected power functions even as the normalization module simplifies to only include `normalize_with_expected_power`. I understand that the models are simple or documented elsewhere but having some examples of how to use them will support the goal of standardized analysis.

I would recommend migrating them to an `expected_power` module or to an example notebook focused on expected power modeling.

Reply:
This is a good question. We are thinking of moving in the direction of keeping modeling functionality out of RdTools. Viewing that more as in the wheelhouse of pvlib and other modeling frameworks. The idea is to keep the open source PV space as clean and organized as possible. But I'm hearing that you see value in some modeling capability within RdTools. I'm happy to keep that discussion open with the broader community.